iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -183,7 +183,7 @@ internal MqttTransportHandler(PipelineContext context, IotHubClientMqttSettings
             _mqttClient = mqttFactory.CreateMqttClient();
             _mqttClientOptionsBuilder = new MqttClientOptionsBuilder();
 
-            _hostName = context.IotHubConnectionCredentials.IotHubHostName;
+            _hostName = context.IotHubConnectionCredentials.HostName;
```

Comment:
👍🏽 
This can be either the hub name directly or a gateway name, right?

Reply:
correct.

`HostName = gatewayHostName ?? hostName;`